src/adaptors/http2/http2_adaptor.c

```diff
@@ -1251,7 +1253,12 @@ static int on_frame_recv_callback(nghttp2_session *session,
                     qdr_delivery_remote_state_updated(http2_adaptor->core, stream_data->out_dlv,
                                                       stream_data->out_dlv_local_disposition, true, 0, false);
                 }
-                if (stream_data->in_dlv) {
+                if (stream_data->in_dlv && !stream_data->in_dlv_decrefed) {
```

Comment:
What if (stream_data->in_dlv && stream_data->in_dlv_decrefed) is true?  Shouldn't stream_data->in_dlv be set to zero in that case since stream_data->in_dlv is an invalid pointer?